docs/scripts/reportBrokenLinks.js

```diff
@@ -95,17 +95,17 @@ const getMdFilesImported = (jsPageFile) => {
   const fileContent = fse.readFileSync(jsPageFile, 'utf8');
   /**
    * Content files can be represented by either:
-   * - 'docsx/data/advanced-components/overview.md?@mui/markdown'; (for mui-x)
-   * - 'docs/data/advanced-components/overview.md?@mui/markdown';
-   * - './index.md?@mui/markdown';
+   * - 'docsx/data/advanced-components/overview.md?muiMarkdown'; (for mui-x)
+   * - 'docs/data/advanced-components/overview.md?muiMarkdown';
+   * - './index.md?muiMarkdown';
    */
-  const importPaths = fileContent.match(/'.*\?@mui\/markdown'/g);
+  const importPaths = fileContent.match(/'.*\?muiMarkdown'/g);
```

Comment:
Which repositories use `?@mui/markdown` still? I was under the impression we replaced it everywhere. 


First time I see this code but as a suggestion: I feel I think it could make more sense to search for an import to a `.mdx?` file instead and ignore the url query altogether:

```tsx
const match = /import .* from \'(.+.mdx?)(\?.*)?\'/.exec(filecontent)
const mdPath = match[1]
```

I think it beter covers the intent. `?@muiMarkdown` is a MUI proprietary thing and I'd rather see us remove it in favor for just a bare `.md` specifier.

Reply:
> Which repositories use ?@mui/markdown still? 

I have configured VS Code to have all the repositories for search

<img width="578" alt="SCR-20240828-dhro" src="https://github.com/user-attachments/assets/629e598b-22be-43a0-b9de-f16091662102">

Yes, I was thinking of Base UI.

> as we're most likely moving away from this markdown loader in favor of MDX.

To make sure we can continue to do extra processing at built-time before this reaches the client bundle. Also maintenance looks a bit dead, so to be careful that this solves problems we are facing https://github.com/mdx-js/mdx/graphs/contributors.

Reply:
> make sure we can continue to do extra processing at built-time before this reaches the client bundle

Sure, nothing has changed in this regard.

> Also the maintenance of MDX looks a bit dead https://github.com/mdx-js/mdx/graphs/contributors, so to be careful that this solves more problems we are facing than it creates 😄.

Interesting. Considering their number of downloads (the same order of magnitude as Material UI), I'm surprised how few issues they have. Perhaps it just works and doesn't need a lot of maintenance.

Reply:
@michaldudak As long as we can implement stuff like https://github.com/mui/material-ui/pull/43562 happy.

> Interesting. Considering their number of downloads (the same order of magnitude as Material UI), I'm surprised how few issues they have. Perhaps it just works and doesn't need a lot of maintenance.

What I see: with a custom stack we can quickly iterate on it to make changes, with MDX, it iteration cycles might be slower. So if we trade speed, I imagine it's because we want other features in exchange. I'm not familiar with what the advantages of MDX are.

Reply:
MDX works as a series of AST transformations. Some(/all?) of those transformations can be manipulated with plugins. I had a chance to play around with this a bit in https://github.com/Janpot/rsc-playground/blob/master/src/lib/remarkPlugin.mjs.

It may potentially have a bit higher learning curve, but it feels more powerful than what we do.

Reply:
> As long as we can implement stuff like https://github.com/mui/material-ui/pull/43562 outside of the bundle we send to browsers

Yup, this shouldn't be a problem at all.

> Some(/all?) of those transformations can be manipulated with plugins.

I actually implemented a [plugin](https://github.com/michaldudak/base-ui/blob/682f3dba1234d0e48c213374d8439c9bca8a93a4/docs/scripts/rehypeExtractLinkUrls.mts) to extract `href` attributes of all links for the updated broken link validator - it's pretty easy (as long as you're familiar with traversing ASTs).